core/src/main/java/org/elasticsearch/node/InternalSettingsPreparer.java

```diff
@@ -112,6 +118,9 @@ public static Environment prepareEnvironment(Settings input, Terminal terminal,
             throw new SettingsException("multiple settings files found with suffixes: "
                 + Strings.collectionToDelimitedString(foundSuffixes, ","));
         }
+        if (foundSuffixes.contains(".yml") || foundSuffixes.contains(".json")) {
+            deprecationLogger.deprecated("elasticsearch.yaml and elasticsearch.json are deprecated. Rename your configuration file to elasticsearch.yaml.");
```

Comment:
Because we disallow multiple settings files (a few lines up), we are certain here that `foundSuffixes` contains at most one element. So we can write:

```java
if (foundSuffixes.contains(".yml") || foundSuffixes.contains(".json")) {
    final DeprecationLogger deprecationLogger = new DeprecationLogger(Loggers.getLogger(InternalSettingsPreparer.class));
    deprecationLogger.deprecated(
        "elasticsearch[{}] is deprecated; rename your configuration file to elasticsearch.yaml",
        foundSuffixes.iterator().next());
}
```

I think this is clean, and let's us scope the logger locally so we can avoid the static fields.